File: DSA/Strings/py/isPalindrome.py
"""
def is_palindrome(str):
  check = []
  for i in reversed(range(len(str))):
    check.append(str[i])


  if(''.join(check) == str):
    return True
  else:
    return False


"""

def is_palindrome(str):
  i=0
  j=len(str)-1
  while(i <= j):
    if str[i] != str[j]:
      return False
    else:
      i+=1
      j-=1

  return True
print(is_palindrome('defdef'))
print(is_palindrome('aabaa'))

File: DSA/Strings/JS/isPalindrome.js
// function isPalindrome(str) {
//   const check = [];

//   for (let i = str.length - 1; i > 0; i--) {
//     check.push(str[i]);
//   }

//   if (str.join('') === convertedStr) return true;
//   else return false;
// }

function isPalindrome(str) {
  let i = 0;
  let j = str.length - 1;

  while (i <= j) {
    if (str[i] !== str[j]) {
      return false;
    } else {
      i++;
      j--;
    }
  }

  return true;
}

console.log(isPalindrome('aabaa'));
console.log(isPalindrome('defdef'));
